sdk/core/switcher/entry.S

```diff
@@ -880,8 +933,21 @@ __Z22switcher_recover_stackv:
 	// two instructions.
 	cspecialr          ca0, mtdc
 	clhu               a1, TrustedStack_offset_frameoffset(ca0)
+	addi               a1, a1, -TrustedStackFrame_size
 	cincoffset         ca0, ca0, a1
 	clc                ca0, TrustedStackFrame_offset_csp(ca0)
+	// If this is the first frame, then the recovered stack will be the stack
+	// on entry.  If this is not the first frame then then we need
```

Comment:
Please complete this comment. A bit more explanation of the lines below would be nice too. Why are these the correct bounds?

Reply:
Done.